Src/DependencyCollector/Shared/Implementation/SqlClientDiagnostics/SqlClientDiagnosticSourceListener.cs

```diff
@@ -87,408 +113,427 @@ public void Dispose()
                 switch (evnt.Key)
                 {
                     case SqlBeforeExecuteCommand:
-                    {
-                        var operationId = (Guid)CommandBefore.OperationId.Fetch(evnt.Value);
-
-                        DependencyCollectorEventSource.Log.SqlClientDiagnosticSubscriberCallbackCalled(operationId, evnt.Key);
-
-                        var command = (SqlCommand)CommandBefore.Command.Fetch(evnt.Value);
-
-                        if (this.operationHolder.Get(command) == null)
+                    case SqlMicrosoftBeforeExecuteCommand:
                         {
-                            var dependencyName = string.Empty;
-                            var target = string.Empty;
+                            var operationId = (Guid)CommandBefore.OperationId.Fetch(evnt.Value);
```

Comment:
if customer happen to have both System.Data.SqlClient and Ms.Data.SqlClient (which might happen), then the same property fetcher will be used for different events (system.* and ms.*)
This will cause some exception and only one kind of events will be tracked. 

So fetchers should be specific to each event